### PR TITLE
Wave Explosion

### PR DESCRIPTION
**Pull request by Fueredoriku**
An explosion triggered by the base to wipe out most enemies when certain conditions are met:
- Time passing and collecting crystals will charge the base up
- A player dying will remove some of the charge
- When the charge is full, the base will zap every enemy and reset it's charge to 0
- The current charge % is showcased by a purple glow that grows from the bottom to the top of the base crystal

This feature was made to create a little bit of breathing room as the enemy waves starts to become more crowded, in addition to rewarding crystal collection and further punishing players for dying. (Due to the latter, I would recommend trying this branch with feature/minor-polish when testing balancing, as that was what I balanced it for)

### CODE REVIEW

**Review by SondreHus — APPROVED** [review]
TLDR 
https://giphy.com/gifs/QDRJ6IJzFSR1K
